src/plugins/intel_gpu/include/intel_gpu/graph/program.hpp

```diff
@@ -343,6 +350,8 @@ struct program {
     // old_node - node which will be replaced
     // new_node - node which will replace the old one
     void replace(program_node& old_node, program_node& new_node);
+
+    void prepare_tools();
```

Comment:
How about to rename this to be more specific, such as prepare_compilation_context ?

Reply:
prepare_tools create not only compilation_context but also kernels_cache, impl_cache, and pass_manager. and they are not all compilation_context. but I agree your opinion too. I will take a time to create more specific name for this function.

Reply:
Done. please review it again